### PR TITLE
Make protected closeBlock and appendTypeParameters in XbaseCompiler.

### PR DESCRIPTION
**Pull request by gallandarakhneorg**
In order to enable overriding of the functions of `XbaseCompiler`, these two functions must be switched to the `protected` visibility.

Signed-off-by: Stéphane Galland <galland@arakhne.org>

### CODE REVIEW

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java [hunk 1]
```diff
@@ -1397,7 +1398,11 @@ protected void _toJavaIfStatement(
 		}
 	}
 
-	private void closeBlock(ITreeAppendable caseAppendable) {
+	/** Close a block of code.
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java]: Please follow the canonical style of writing JavaDoc. That would be a line break after the second star:
```
/**
 * Close the block of code.
 * ...
 */
```

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java [hunk 2]
```diff
@@ -1397,7 +1398,11 @@ protected void _toJavaIfStatement(
...
+	 *
+	 * @param caseAppendable the receiver of the block closing.
+	 */
+	protected void closeBlock(ITreeAppendable caseAppendable) {
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java]: Please rename the parameter if you make this accessible.

**Comment by gallandarakhneorg** [comment]
Changes are applied according to your queries.

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java [hunk 3]
```diff
@@ -1707,28 +1713,35 @@ protected boolean needSyntheticSelfVariable(XClosure closure, LightweightTypeRef
...
+	 *
+	 * @param codeReceiver the receiver of the Java code.
+	 * @param operation the source operation.
+	 * @param instantiatedType the type of the operation container.
+	 */
+	protected void appendTypeParameters(ITreeAppendable codeReceiver, JvmOperation operation, LightweightTypeReference instantiatedType) {
 		BottomResolvedOperation resolvedOperation = new BottomResolvedOperation(operation, instantiatedType, overrideTester);
 		List<JvmTypeParameter> typeParameters = resolvedOperation.getResolvedTypeParameters();
-		b.append("<");
+		codeReceiver.append("<");
```
[the remark this answers is not in the document]
Comment (cdietrich) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java]: :+1: for the type check and renaming to appendable

**Comment by cdietrich** [comment]
thx @gallandarakhneorg

**Pull request merged**